src/algorithms/cache.rs

```diff
@@ -0,0 +1,280 @@
+// Where is the cut-off?
+const X0: f64 = 0.00000497;
+// This combination of settings leads to a fairly sharp cut-off around:
+//
+//  word  "raw" p      sigmoid p   count
+// vying 0.000491% -> 15.999853% (1560905)
+// rheum 0.000492% -> 16.735476% (1561474)
+// lamas 0.000492% -> 16.827817% (1561544)
+// kraal 0.000492% -> 17.389054% (1561963)
+// gents 0.000493% -> 25.448008% (1567081)
+// hails 0.000494% -> 29.575062% (1569275)
+// atria 0.000494% -> 30.602258% (1569792)
+// wooed 0.000495% -> 32.304510% (1570628)
+// kinks 0.000495% -> 32.904357% (1570917)
+// sushi 0.000495% -> 34.261053% (1571561)
+// filly 0.000496% -> 39.634216% (1574006)
+// lazar 0.000496% -> 43.073524% (1575508)
+// lapel 0.000497% -> 48.215265% (1577704)
+// cecum 0.000497% -> 48.505414% (1577827)
+// kayak 0.000497% -> 49.307915% (1578167)
+// fundy 0.000498% -> 55.755065% (1580908)
+// haply 0.000498% -> 56.508662% (1581232)
+// bigot 0.000498% -> 57.402526% (1581618)
+// wisps 0.000498% -> 57.407146% (1581620)
+// foals 0.000498% -> 57.875326% (1581823)
+fn sigmoid(p: f64) -> f64 {
+    L / (1.0 + (-K * (p - X0)).exp())
+}
+const PRINT_SIGMOID: bool = false;
+
+impl Cache {
+    pub fn new() -> Self {
+        let remaining = Cow::Borrowed(INITIAL.get_or_init(|| {
+            let mut sum = 0;
+            let mut words = Vec::from_iter(DICTIONARY.lines().map(|line| {
+                let (word, count) = line
+                    .split_once(' ')
+                    .expect("every line is word + space + frequency");
+                let count: usize = count.parse().expect("every count is a number");
+                sum += count;
+                (word, count)
+            }));
+
+            words.sort_unstable_by_key(|&(_, count)| std::cmp::Reverse(count));
+
+            if PRINT_SIGMOID {
+                for &(word, count) in words.iter().rev() {
+                    let p = count as f64 / sum as f64;
+                    println!(
+                        "{} {:.6}% -> {:.6}% ({})",
+                        word,
+                        100.0 * p,
+                        100.0 * sigmoid(p),
+                        count
+                    );
+                }
+            }
+
+            let words: Vec<_> = words
+                .into_iter()
+                .enumerate()
+                .map(|(idx, (word, count))| (word, sigmoid(count as f64 / sum as f64), idx))
+                .collect();
+
+            words
+        }));
+
+        let dimension = remaining.len();
+
+        Self {
+            remaining,
+            patterns: Cow::Borrowed(PATTERNS.get_or_init(|| Correctness::patterns().collect())),
+            entropy: Vec::new(),
+            computes: COMPUTES
+                .get_or_init(|| (dimension, vec![Default::default(); dimension * dimension])),
+        }
+    }
+}
+
+fn cachable_enumeration(answer: &str, guess: &str) -> u8 {
+    enumerate_mask(&Correctness::compute(answer, guess)) as u8
+}
+
+fn get_row(
```

Comment:
How about `packed_correctnesses_for`? Also, with the `build.rs` generating the array this doesn't need to take `computes` since it can just use the static.